agcounts/legacy.py

```diff
@@ -0,0 +1,231 @@
+        bpf_data[0, 0] = zeros_comp - poles_comp
+        shift_reg_out[[0], 1:9] = shift_reg_out[[0], 0 : (9 - 1)]
+        shift_reg_out[0, 0] = zeros_comp - poles_comp
+
+    for j in range(len(bpf_data[0])):
+        shift_reg_in[[0], 1:9] = shift_reg_in[[0], 0:8]
+        shift_reg_in[0, 0] = down_sample_data[0, j]
+        zeros_comp = np.sum(INPUT_COEFFICIENTS[[0], 0:8] * shift_reg_in[[0], 0:8])
+        poles_comp = np.sum(OUTPUT_COEFFICIENTS[[0], 1:8] * shift_reg_out[[0], 0:7])
+        bpf_data[0, j] = zeros_comp - poles_comp
+        shift_reg_out[[0], 1:9] = shift_reg_out[[0], 0 : (9 - 1)]
+        shift_reg_out[0, 0] = zeros_comp - poles_comp
+    del down_sample_data
+    gc.collect(1)
+
+    bpf_data = (
+        (3.0 / 4096.0) / (2.6 / 256.0) * 237.5
+    ) * bpf_data  # 17.127404 is used in ActiLife and 17.128125 is used in firmware.
+
+    logger.info("Threshold/trimming data")
+
+    # then threshold/trim
+    trim_data = np.zeros((1, len(bpf_data[0])))  # type: ignore
+
+    if lfe_select:
+        min_count = 1
+        max_count = 128 * 1
+
+        for i in range(len(bpf_data[0])):  # type: ignore
+            if abs(bpf_data[0, i]) > max_count:  # type: ignore
+                trim_data[0, i] = max_count
+            elif abs(bpf_data[0, i]) < min_count:  # type: ignore
+                trim_data[0, i] = 0
+            elif abs(bpf_data[0, i]) < 4:  # type: ignore
+                trim_data[0, i] = np.floor(abs(bpf_data[0, i])) - 1  # type: ignore
+            else:
+                trim_data[0, i] = np.floor(abs(bpf_data[0, i]))  # type: ignore
+    else:
+        min_count = 4
+        max_count = 128
+
+        for i in range(len(bpf_data[0])):  # type: ignore
+            if abs(bpf_data[0, i]) > max_count:  # type: ignore
+                trim_data[0, i] = max_count
+            elif abs(bpf_data[0, i]) < min_count:  # type: ignore
+                trim_data[0, i] = 0
+            else:
+                trim_data[0, i] = np.floor(abs(bpf_data[0, i]))  # type: ignore
+
+    logger.info("Getting data back to 10Hz for accumulation")
+    del bpf_data
+    # hackish downsample to 10 Hz
+    down_sample10_hz = np.zeros((1, int(len(trim_data[0]) / 3)))
+
+    for y in range(1, len(down_sample10_hz[0]) + 1):
+        down_sample10_hz[0, y - 1] = np.floor(
+            np.nanmean(trim_data[0, ((y - 1) * 3) : ((y - 1) * 3 + 3)])
+        )  # floor
+    del trim_data
+    gc.collect(1)
+
+    # Accumulator for epoch
+    block_size = epoch * 10
+    epoch_counts = np.zeros((1, int((len(down_sample10_hz[0]) / block_size))))
+
+    logger.info("Summing epochs")
+    for i in range(len(epoch_counts[0])):
+        epoch_counts[0, i] = np.floor(
+            sum(down_sample10_hz[0, i * block_size : i * block_size + block_size])
+        )
+    del down_sample10_hz
+    gc.collect(1)
+    return epoch_counts
+
+
+def _factors(frequency: int):
+    if frequency == 30:
```

Comment:
Can use a dictionary `frequency -> (upsample_factor, downsample_factor)` defaulting to `(1, 1)` e.g. `dictionary.get(default=(1, 1))`